packages/flutter_tools/lib/src/base/analyze_size.dart

```diff
@@ -0,0 +1,394 @@
+
+          if (path.endsWith('libapp.so')) {
+            childWithPathAsName.name += ' (Dart AOT)';
+          } else if (path.endsWith('libflutter.so')) {
+            childWithPathAsName.name += ' (Flutter Engine)';
+          }
+          currentNode.addChild(childWithPathAsName);
+        }
+        childWithPathAsName.addValue(pathsToSize[paths]);
+        currentNode = childWithPathAsName;
+      }
+      currentNode = rootNode;
+    }
+
+    return rootNode;
+  }
+
+  /// Prints the paths from currentNode all leaf nodes.
+  void _printLibDetails(
+    SymbolNode currentNode,
+    String totalPath,
+    File aotSizeJson,
+  ) {
+    totalPath += currentNode.name;
+
+    if (currentNode.children.isNotEmpty && !currentNode.name.contains('libapp.so')) {
+      for (final SymbolNode child in currentNode.children) {
+        _printLibDetails(child, totalPath + '/', aotSizeJson);
+      }
+    } else {
+      // Print total path and size if currentPath does not have any chilren.
+      _printEntitySize(totalPath, currentNode.value, 2);
+
+      const String libappPath = 'lib/arm64-v8a/libapp.so';
+      // TODO(peterdjlee): Analyze aot size for all platforms.
+      if (totalPath.contains(libappPath)) {
+        _printAotSizeDetails(aotSizeJson);
+      }
+    }
+  }
+
+  /// Go through the AOT gen snapshot size JSON and print out a collapsed summary
+  /// for the first package level.
+  void _printAotSizeDetails(File aotSizeJson) {
+    final SymbolNode root = _parseSymbols(
+      json.decode(aotSizeJson.readAsStringSync()) as List<dynamic>,
+    );
+
+    final int totalSymbolSize = root.children.fold(
+      0,
+      (int previousValue, SymbolNode element) => previousValue + element.value,
+    );
+
+    _printEntitySize(
+      'Dart AOT symbols accounted decompressed size',
+      totalSymbolSize,
+      3,
+    );
+
+    final List<SymbolNode> sortedSymbols = root.children.toList()
+      ..sort((SymbolNode a, SymbolNode b) => b.value.compareTo(a.value));
+    for (final SymbolNode node in sortedSymbols.take(10)) {
+      _printEntitySize(node.name, node.value, 4);
+    }
+  }
+
+  /// Adds breakdown of aot size data as the children of the node at the given path.
+  Map<String, dynamic> addAotSizeDataToJson(
+    Map<String, dynamic> apkAnalysisJson,
+    List<String> path, 
+    List<dynamic> aotSizeJson,
+  ) {
+    Map<String, dynamic> currentLevel = apkAnalysisJson;
+    while (path.isNotEmpty) {
+      final List<Map<String, dynamic>> children = currentLevel['children'] as List<Map<String, dynamic>>;
+      final Map<String, dynamic> childWithPathAsName = children.firstWhere(
+        (Map<String, dynamic> child) => child['n'] as String == path.first,
+      );
+      path.removeAt(0);
+      currentLevel = childWithPathAsName;
+    }
+    currentLevel['children'] = treemapFromJson(aotSizeJson)['children'];
+    return apkAnalysisJson;
+  }
+
+  /// A pretty printer for an entity with a size.
+  void _printEntitySize(
```

Comment:
I'd somehow via the names make it clearer that _printLibDetails prints a tree and _printEntitySize prints a line

Reply:
Done